### PR TITLE
CHG: [droid] use gradle to build apk

### PR DESCRIPTION
**Pull request by koying**
Needed if we want to pull external java build dependencies, eg for upcoming Oreo leanback

### CODE REVIEW

File: tools/android/packaging/Makefile.in [hunk 1]
```diff
@@ -1,5 +1,8 @@
 include ../../depends/Makefile.include
 
+BUILD_TYPE=@CMAKE_BUILD_TYPE@
+BUILD_TYPE_LC = `echo $(BUILD_TYPE) | tr A-Z a-z`
```
Comment (wsnipex) [inline tools/android/packaging/Makefile.in]: BUILD_TYPE_LC := $(shell echo $(BUILD_TYPE) | tr A-Z a-z)
better yet, convert to lc in cmake

File: tools/android/packaging/Makefile.in [hunk 2]
```diff
@@ -153,17 +153,15 @@ libs: $(PREFIX)/lib/@APP_NAME_LC@/lib@APP_NAME_LC@.so
...
-	@$(DX) --dex --output=xbmc/classes.dex xbmc/obj xbmc/lib
 
-package: libs python xbmc/classes.dex
-	@echo "Creating package..."
-	@cp images/@APP_NAME_LC@app-debug-skeleton.apk images/@APP_NAME_LC@app-debug-$(CPU)-unaligned.apk
-	@cd xbmc; $(ZIP) -r -q ../images/@APP_NAME_LC@app-debug-$(CPU)-unaligned.apk lib/$(CPU) assets classes.dex
+package: libs python java
+	@echo "Gradle build..."
+	ANDROID_HOME=$(SDKROOT) ./gradlew assemble$(BUILD_TYPE)Unsigned
+	@cp xbmc/build/outputs/apk/xbmc-$(BUILD_TYPE_LC)Unsigned-unsigned.apk images/@APP_NAME_LC@app-debug-$(CPU)-unaligned.apk
```
Comment (wsnipex) [inline tools/android/packaging/Makefile.in]: do we need to keep xbmc-$(BUILD_TYPE_LC)Unsigned-unsigned.apk or can we just mv to new location?

File: tools/android/packaging/gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#Fri Sep 01 12:45:21 CEST 2017
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-bin.zip
```
Comment (wsnipex) [inline tools/android/packaging/gradle/wrapper/gradle-wrapper.properties]: do we really need to include gradle-wrapper.jar and it's properties or can we use it from the SDK?

**Review by wsnipex — COMMENTED** [review]
looks good, only a minor and the 2 questions

File: tools/android/packaging/Makefile.in (hunk 2 above)
Comment (koying) [reply tools/android/packaging/Makefile.in]: Good question. I actually have very little clue about the gradle inners.
We did a cp before (https://github.com/xbmc/xbmc/pull/13263/files#diff-b3e127cc38edd4b1068565100a34c0e0L165) so I just kept the same logic

File: tools/android/packaging/gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (koying) [reply tools/android/packaging/gradle/wrapper/gradle-wrapper.properties]: Good question as well.
The gradle in SDK is antiquated, though
```
distributionUrl=http\://services.gradle.org/distributions/gradle-1.12-all.zip
```

File: tools/android/packaging/Makefile.in (hunk 1 above)
Comment (koying) [reply tools/android/packaging/Makefile.in]: Convention, or is there something wrong with ``` ` ``` ?

Comment (wsnipex) [reply tools/android/packaging/Makefile.in]: kinda, the backticks method is antiquated and we use $(shell) where possible.
Note the := causes the var to be expanded only once at the beginning, so it's preferred here. (also for the line above btw)

File: tools/android/packaging/gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (koying) [reply tools/android/packaging/gradle/wrapper/gradle-wrapper.properties]: The fact that this gradle wrapper in in a "template" subdir suggests it's not supposed to be used directly, imo

**Comment by koying** [comment]
jenkins build this please

File: tools/android/packaging/gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (wsnipex) [reply tools/android/packaging/gradle/wrapper/gradle-wrapper.properties]: ok

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Review by Rechi — COMMENTED** [review]
this isn't something I've much knowledge about
merge if it works for you

**Pull request merged**